File: js/15.js
/* 
* @Author: ben_cripps
* @Date:   2014-12-07 15:58:20
* @Last Modified by:   ben_cripps
* @Last Modified time: 2014-12-07 16:12:27
*/

'use strict';

function factorial(n) {
    var ret = 1;
    for ( var i = n; i > 1; i-- ) {
        ret*= i
        
    }       
    return ret;
}

function countRoutes(n) {
    var route = factorial(n);
    return ( factorial(2*n) / route * route );
}

console.log(countRoutes(20));

File: python/15.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2015-01-20 20:20:25
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-20 20:27:45

def factorial(n):
    return reduce(lambda x,y: x*y, range(1, n+1))

n = 20
fac_n = factorial(n)

print(factorial(2*n) / (fac_n * fac_n))